### PR TITLE
Refactor: comprehensive unit tests for FileUtils (Deletion, Validation & Storage)

### LINKED ISSUE #4713: Improve code coverage for FileUtils: Deletion, Validation & Storage Logic
**Particular area and files to be tested**
* `core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt`

**Current Status of the tests**
`FileUtils.kt` currently has partial coverage. The existing tests cover `getAllZimParts` (detecting split archives), but critical logic for **file deletion**, **validation**, and **URI resolution** remains untested. This issue aims to achieve >90% coverage by filling these gaps.

**Particular Behaviour to be tested 

1. Deletion Logic
* **Deletion:** Verify `deleteZimFile` correctly removes the main `.zim` file and any associated `.part` or split-archive chunks (`.zimaa`, etc.).
* **Part Cleanup:** Verify logic that detects and deletes temporary `.part` files when the main file is deleted.
* **Filename Resolution:** Verify `getFileName` correctly resolves the target path (handling collision between `.zim` and `.part` extensions).

2. ZIM Validation 
* **Extension Checks:** Verify `isValidZimFile` validates `.zim`, `.zimaa`, and `.zim.part` extensions while rejecting invalid files.
* **Split Identification:** Verify `isSplittedZimFile` correctly identifies multi-volume archive parts (e.g., `.zimaa`, `.zimab`).
* **File Descriptors:** Verify `isFileDescriptorCanOpenWithLibkiwix` to prevent runtime crashes with invalid descriptors.

3. URI & Storage Resolution
* **URI Resolution:** Verify `getLocalFilePathByUri` correctly resolves local paths for `file://` and `content://` URIs (Downloads/External Storage).
* **Storage Volumes:** Verify `getStoragePath` correctly resolves paths for "primary" internal storage vs. SD cards.
* **Edge Cases:** Verify handling of null URIs, HTTP schemes, and unknown storage volumes.

### PR DESCRIPTION
**Pull request by harshsomankar123-tech**
fixes #4713 
 Summary
This PR significantly improves the code coverage of `FileUtils.kt` by adding comprehensive unit tests for critical file operations, ZIM validation, and storage path resolution. It achieves **>90% coverage** for the file.

 Changes
* **Preserved Existing Tests:** Kept the original tests for `getAllZimParts` to ensure no regression in split-archive detection.
* **Added Deletion Logic:** Added tests for `deleteZimFile` to verify correct removal of `.zim`, `.part`, and split chunks (`.zimaa`, etc.).
* **Added Validation Logic:** Added tests for `isValidZimFile` and `isSplittedZimFile` to validate file extensions and types.
* **Added Storage & URI Logic:** Added tests for `getStoragePath` and `getLocalFilePathByUri` using `MockK` to mock Android static classes (`Environment`, `Uri`, `Context`).

Technical Details
* Used `mockkStatic` to mock Kotlin extension functions and `Android.os.Environment`.
* Covered edge cases for file collision (e.g., when both `book.zim` and `book.zim.part` exist).
* Verified safety checks for file descriptors to prevent runtime crashes.

### CODE REVIEW

**Comment by atharvyadav22** [comment]
@harshsomankar123-tech i have recently changed File Utils please view that and add tests accordingly as i have made new features

**Comment by harshsomankar123-tech** [comment]
@atharvyadav22 Sure, I’ll update the tests accordingly

**Comment by harshsomankar123-tech** [comment]
@MohitMaliFtechiz could you please review this when you get time? Thanks!

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 1]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+    val result = FileUtils.getSafeFileNameAndSourceFromUrlOrSrc(null, "https://example.com/src.png")
+    assertEquals("src.png", result?.first)
+    assertEquals("https://example.com/src.png", result?.second)
+
+    unmockkStatic(android.webkit.URLUtil::class)
+  }
+
+  @Test
+  fun getDownloadRootDir_returnsExternalMediaDirs_forPlayStoreBuildWithAndroid11OrAbove() = runBlocking {
+    val kiwixDataStore = mockk<org.kiwix.kiwixmobile.core.utils.datastore.KiwixDataStore>()
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt]: Fix the formatting issues.

Image

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 2]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+
+  @Test
+  fun deleteZimFile_deletesZimFileAndChecksPartFiles() = runBlocking {
+    val path = "/storage/emulated/0/wiki.zim"
+    coEvery { any<File>().deleteFile(any()) } returns true
+    coEvery { any<File>().isFileExist(any()) } coAnswers {
+      val p = firstArg<File>().path
+      p == "$path.part"
+    }
+    FileUtils.deleteZimFile(path)
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt:228]: This only deletes the file, but does not check that the file is actually deleted or not. We should improve this test case.

```kotlin
@Test
  fun deleteZimFile_deletesZimFileAndChecksPartFiles() = runBlocking {
    val path = "/storage/emulated/0/wiki.zim"
    val deletedPaths = mutableListOf<String>()
    coEvery { any<File>().deleteFile(any()) } coAnswers {
      deletedPaths.add(firstArg<File>().path)
      true
    }
    coEvery { any<File>().isFileExist(any()) } coAnswers {
      val p = firstArg<File>().path
      p == "$path.part"
    }
    FileUtils.deleteZimFile(path)
    assertTrue(deletedPaths.contains(path))
    assertTrue(deletedPaths.any { it.contains(".part") })
  }
```

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 3]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+  }
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt:249]: We should add a test for multiple chunks deletion.

```kotlin
 @Test
  fun deleteZimFile_deletesMultipleChunks_sequentially() = runBlocking {
    val basePath = "/storage/emulated/0/wiki.zim"
    val inputPath = "${basePath}aa"
    val deleted = mutableListOf<String>()
    coEvery { any<File>().deleteFile(any()) } coAnswers {
      deleted.add(firstArg<File>().path)
      true
    }
    coEvery { any<File>().isFileExist(any()) } coAnswers {
      val p = firstArg<File>().path
      p == "${basePath}aa" || p == "${basePath}ab"
    }

    FileUtils.deleteZimFile(inputPath)
    assertTrue(deleted.any { it.contains("aa") })
    assertTrue(deleted.any { it.contains("ab") })
  }
```

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 4]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+
+  @Test
+  fun decodeBase64DataUri_returnsExtensionAndBytes_forValidBase64() {
+    val base64Data = "mocked_base_64_encoded_string"
+    val input = "data:image/png;base64,$base64Data"
+    val expectedBytes = byteArrayOf(1, 2, 3)
+
+    mockkStatic(android.webkit.MimeTypeMap::class)
+    val mockMimeTypeMap = mockk<android.webkit.MimeTypeMap>()
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt]: Do not import with the full package name. Remove all the full package name imports.

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 5]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+    every { android.util.Base64.decode(base64Data, android.util.Base64.DEFAULT) } returns expectedBytes
+
+    val result = FileUtils.decodeBase64DataUri(input)
+    assertEquals("png", result?.first)
+    assertTrue(result?.second?.contentEquals(expectedBytes) == true)
+
+    unmockkStatic(android.webkit.MimeTypeMap::class)
+    unmockkStatic(android.util.Base64::class)
+  }
+
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt:445]: We should add the edge case as well.


```kotlin
@Test
  fun decodeBase64DataUri_returnsBin_whenMimeUnknown() {
    val base64 = "data:unknown/type;base64,SGVsbG8="

    mockkStatic(MimeTypeMap::class)
    val mockMime = mockk<MimeTypeMap>()
    every { MimeTypeMap.getSingleton() } returns mockMime
    every { mockMime.getExtensionFromMimeType(any()) } returns null

    val result = FileUtils.decodeBase64DataUri(base64)

    assertEquals("bin", result?.first)

    unmockkStatic(MimeTypeMap::class)
  }
```

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt [hunk 6]
```diff
@@ -102,4 +125,325 @@ class FileUtilsTest {
...
+    every { android.webkit.URLUtil.guessFileName(null, null, null) } returns "downloadfile.bin"
+    every { android.webkit.URLUtil.guessFileName("https://example.com/src.png", null, null) } returns "src.png"
+
+    val result = FileUtils.getSafeFileNameAndSourceFromUrlOrSrc(null, "https://example.com/src.png")
+    assertEquals("src.png", result?.first)
+    assertEquals("https://example.com/src.png", result?.second)
+
+    unmockkStatic(android.webkit.URLUtil::class)
+  }
+
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/files/FileUtilsTest.kt:483]: We should add edge cases as well.

```kotlin
@Test
  fun getSafeFileNameAndSourceFromUrlOrSrc_returnsNull_whenBothInputsNull() {
    val result = FileUtils.getSafeFileNameAndSourceFromUrlOrSrc(null, null)
    assertEquals(null, result)
  }

  @Test
  fun getSafeFileNameAndSourceFromUrlOrSrc_returnsNull_forBinFile() {
    mockkStatic(URLUtil::class)
    every { URLUtil.guessFileName(any(), any(), any()) } returns "file.bin"

    val result =
      FileUtils.getSafeFileNameAndSourceFromUrlOrSrc("https://example.com/file.bin", null)

    assertEquals(null to "https://example.com/file.bin", result)

    unmockkStatic(URLUtil::class)
  }
```

**Comment by harshsomankar123-tech** [comment]
@MohitMaliFtechiz Please take a look. The CI failures will be addressed in  #4780 .
